src/graphnet/data/dataset.py

```diff
@@ -30,6 +29,9 @@ def __init__(
         string_selection: Optional[List[int]] = None,
         selection: Optional[List[int]] = None,
         dtype: torch.dtype = torch.float32,
+        loss_weight_table: str = None,
+        loss_weight_column: str = None,
+        loss_weight_padding_value: float = 1.0,
```

Comment:
It seems that `loss_weight_padding_value` is only used if `loss_weight_column is None`, i.e. as a default value. Given that `LossFunction.forward` already has a sensible default behaviour (all samples weights are one when `weights=None` is passed) I suggest we remove the default-behaviour-handling code from here so as to avoid duplication. My suggestions below reflect this.
```suggestion
```

Reply:
Not entirely. The padding value is assigned as weight for events that has no weights in the weight table. So if the query is empty.

Reply:
Brilliant, following our discussion there are two potential use cases/options:

1. You either specify a sample weight for all events in your dataset. In this case a missing weight would be a sign of an error, and we would want to alert the user to this.
2. You knowingly only specify a sample weight for a subset of your data (e.g., only for neutrino MC events when training on a mix of neutrino MC and real data). In this case a missing weight is expected and not _necessarily_ a sign of an error.

I would always recommend option (1.), but I acknowledge that it might be useful to leave option (2.) open for those that are brave enough to implicitly ignore potential missing weights.

In this case, I thing we should rename the variable (I understand "padding" to be something different, at least) and make it optional. If the variable is `None` and missing weights are found, this should raise an exception. This ought to be the default behaviour. If the variable is not `None`, but instead a float, this number should be used as the weight for all events with a "missing" entry in the weight column.

```suggestion
        loss_weight_default_value: Optional[float] = None,
```